public/static/docs/api-reference/summon.md

```diff
@@ -0,0 +1,31 @@
+# dvc.api.summon()
+
+Instantiate an object, described in a _summon file_.
```

Comment:
What's the best place for me to read what exactly is a summon file and what it's YAML structure is? I think we will need a new doc in the user guide to explain it, and link from here... Cc @Suor

Reply:
Is this the best ref rn? https://github.com/iterative/dvc/issues/2719#issuecomment-549856613 Lots of comments below this one, not sure which structure and behavior is the latest.

Reply:
I think I'm going to extract this part about `dvc.api.summon` details into a separate issue so we can merge a first version of the API ref.

Reply:
See #927